README.md

```diff
@@ -74,97 +79,92 @@ df.columns = ['x', 'y', 'z1']
 
-To run a specific doctest/unittest:
-```bash
-# Run unittest
-./dev/pytest -k "DataFrameTest and test_Dataframe"
+- Koalas docs' style and infrastructure simply follow rest of the PyData projects'.
 
-# Run doctest
-./dev/pytest -k DataFrame.corr --doctest-modules databricks
-```
+### Unify small data (pandas) API and big data (Spark) API, but pandas first
 
-Note that `-k` is used for simplicity although it takes an expression. You can use `--verbose` to check what to filter. See `pytest --help` for more details.
+The Koalas DataFrame is meant to provide the best of pandas and Spark under a single API, with easy and clear conversions between each API when necessary. When Spark and pandas have similar APIs with subtle differences, the principle is to honor the contract of the pandas API first.
 
-### Building Documentation
+There are 4 different classes of functions:
 
-To build documentation via Sphinx:
+ 1. Functions that are only found in Spark (`select`, `selectExpr`). These functions should also be available in Koalas.
 
-```bash
-cd docs && make clean html
-```
+ 2. Functions that are found in Spark but that have a clear equivalent in pandas, e.g. `alias` and `rename`. These functions will be implemented as the alias of the pandas function, but should be marked that they are aliases of the same functions. They are provided so that existing users of PySpark can get the benefits of Koalas without having to adapt their code.
 
-It generates HTMLs under `docs/build/html` directory. Open `docs/build/html/index.html` to check if documentation is built properly.
+ 3. Functions that are found in both Spark and pandas under the same name (`count`, `dtypes`, `head`). The return value is the same as the return type in pandas (and not Spark's).
 
-### Coding Conventions
-We follow [PEP 8](https://www.python.org/dev/peps/pep-0008/) with one exception: lines can be up to 100 characters in length, not 79.
+ 4. Functions that are only found in pandas. When these functions are appropriate for distributed datasets, they should become available in Koalas.
 
-### Release Instructions
-Only project maintainers can do the following.
+### Return Koalas data structure for big data, and pandas data structure for small data
 
-Step 1. Make sure version is set correctly in `databricks/koalas/version.py`.
+Often developers face the question whether a particular function should return a Koalas DataFrame/Series, or a pandas DataFrame/Series. The principle is: if the returned object can be large, use a Koalas DataFrame/Series. If the data is bound to be small, use a pandas DataFrame/Series. For example, `DataFrame.dtypes` return a pandas Series, because the number of columns in a DataFrame is bounded and small, whereas `DataFrame.head()` or `Series.unique()` returns a Koalas DataFrame, because the resulting object can be large.
```

Comment:
> whereas `DataFrame.head()` or `Series.unique()` returns a Koalas DataFrame, because the resulting object can be large.

typos ..

whereas `DataFrame.head()` or `Series.unique()` returns a Koalas DataFrame/Series, because the resulting object can be large.